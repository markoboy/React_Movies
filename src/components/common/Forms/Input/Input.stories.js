import React from 'react';

import withStoryBackgroundWrapper from '@components/hocs/WithStoryBackgroundWrapper';
import Input from './index';

export default {
  component: Input,
  title: 'Common/Forms/Input',
  decorators: [withStoryBackgroundWrapper],
};

const Template = (args) => <Input {...args} />;

export const Default = Template.bind({});
Default.args = {
  placeholder: 'Form input',
};

export const WithLabel = Template.bind({});
WithLabel.args = {
  label: 'Form input label',
  placeholder: 'Form input',
};
