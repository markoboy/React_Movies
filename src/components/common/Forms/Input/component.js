import withFormElementWrapper from '@components/hocs/WithFormElementWrapper';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import React, { useCallback, useMemo } from 'react';
import FormLabelComponent from '../FormLabel';
import './styles.scss';

function InputComponent({
  type,
  placeholder,
  name,
  id,
  classes,
  value,
  onChange,
  onClick,
  label,
  required,
  pattern,
  disabled,
  children
}) {
  const className = useMemo(() => clsx('form__input', classes), [classes]);

  const handleChange = useCallback((event) => onChange && onChange(event.target.value), [onChange]);

  return (
    <>
      <FormLabelComponent label={label} id={id} />
      <input
        className={className}
        type={type}
        placeholder={placeholder}
        name={name}
        id={id}
        value={value}
        onChange={handleChange}
        onClick={onClick}
        required={required}
        pattern={pattern}
        disabled={disabled}
      />
      {children}
    </>
  );
}

InputComponent.defaultProps = {
  type: 'text',
  placeholder: '',
  name: '',
  id: '',
  classes: '',
  value: '',
  onChange: null,
  onClick: null,
  label: '',
  required: false,
  pattern: null,
  disabled: false,
};

InputComponent.propTypes = {
  type: PropTypes.string,
  placeholder: PropTypes.string,
  name: PropTypes.string,
  id: PropTypes.string,
  classes: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onChange: PropTypes.func,
  onClick: PropTypes.func,
  label: PropTypes.string,
  required: PropTypes.bool,
  pattern: PropTypes.string,
  disabled: PropTypes.bool,
};

export default withFormElementWrapper(InputComponent);
