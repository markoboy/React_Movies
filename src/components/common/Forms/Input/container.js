import clsx from 'clsx';
import React, { memo, useMemo } from 'react';
import InputComponent from './component';

function InputContainer({ classes, children, inputRef, ...props }) {
  const className = useMemo(() => clsx('form__input', classes), [classes]);

  return (
    <InputComponent classes={className} inputRef={inputRef} {...props}>
      {children}
    </InputComponent>
  );
}

InputContainer.defaultProps = InputComponent.defaultProps;

InputContainer.propTypes = InputComponent.propTypes;

export default memo(InputContainer);
