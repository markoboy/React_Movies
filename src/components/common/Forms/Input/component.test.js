import React from 'react';
import { render } from '@testing-library/react';

import Input from './component';

const mockFn = jest.fn();

describe('Input Component', () => {
  it('renders with provided text', () => {
    const { container, getByText } = render(
      <Input onChange={mockFn}>Input component</Input>
    );

    const input = getByText(/Input component/);

    expect(input).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
