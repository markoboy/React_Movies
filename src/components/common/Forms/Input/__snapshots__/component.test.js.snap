// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders with provided text 1`] = `
<div>
  <div
    class="form-element-wrapper"
  >
    
    <input
      type="text"
      value=""
    />
    Input component
  </div>
</div>
`;
